tests/unit/fem/test_assembler_restriction.py

```diff
@@ -517,7 +517,7 @@ def test_vector_assembly_with_restriction(
     assert_vector_equal(unrestricted_vector_nonlinear, restricted_vector_nonlinear, dofmap_restriction)
     unrestricted_fem_module.petsc.set_bc(unrestricted_vector_nonlinear, bcs, unrestricted_solution)
     restricted_fem_module.petsc.set_bc(
-        restricted_vector_nonlinear, bcs, restricted_solution, restriction=dofmap_restriction)
+        restricted_vector_nonlinear, bcs, restricted_solution, restriction=dofmap_restriction, restriction_x0=dofmap_restriction)
```

Comment:
I am a bit confused. Wasn't the third argument now supposed to become the `unrestricted_solution`? How come that this still works?

Reply:
The third argument is the solution, unrestricted or not, and the last named argument here is the associated restriction

Reply:
> The third argument is the solution, unrestricted or not.

I'll try have a stab at clarify this in the documentation. I would try a concise way to say that this must always be the unrestricted solution: if you then pass in `restriction_x0`, then it will get restricted. If you do not pass in `restriction_x0`, it will get used as it is.

If this is true, then the test is actually wrong, as I was expecting `unrestricted_solution` to appear instead of `restricted_solution`. I am surpirsed the test still passes on your local copy.